src/main/java/net/greghaines/jesque/worker/Watchdog.java

```diff
@@ -124,6 +143,40 @@ private void activateIsAliveService(final String serverName) throws RuntimeExcep
         scheduler.scheduleAtFixedRate(isAlive, INITIAL_DELAY, IS_ALIVE_PERIOD, SECONDS);
     }
 
+    private void heartbeatIsAlive(final Jedis jedis) {
+        String recoveringStatus = getRecoveryStatus();
+
+        final String needRedisRecovery = (String) jedis.evalsha(watchdogScriptHash.get(), WATCHDOG_SCRIPT_KEYS_NUMBER, serverName, IS_ALIVE, getCurrTime(), isDebug, recoveringStatus , String.valueOf(LIGHT_KEEPER_PERIOD));
+
+
+        if (Boolean.valueOf(needRedisRecovery)) {
+            recovering = recoverFromRedisRestart();
+        }
+    }
+
+    private String getRecoveryStatus() {
```

Comment:
Might be nicer to return the enum and transform to String only when you actually need it.
It will cause you to write the `.name()` part only once.